compiler/qsc_frontend/src/resolve.rs

```diff
@@ -569,6 +582,68 @@ impl Resolver {
                 scope.tys.insert(Rc::clone(&name.name), id);
                 scope.terms.insert(Rc::clone(&name.name), id);
             }
+            ast::ItemKind::Export(export) => {
+                // resolve the exported item and insert the vec ident into the names table, so we can access it in
+                // lowering
+                for item in export.items() {
+                    let resolved_item = match resolve(
+                        NameKind::Term,
+                        &self.globals,
+                        self.locals.get_scopes(&self.curr_scope_chain),
+                        &item.name,
+                        &item.namespace,
+                    ) {
+                        Ok(res) => res,
+                        Err(_) => {
+                            // try to see if it is a type
+                            match resolve(
+                                NameKind::Ty,
+                                &self.globals,
+                                self.locals.get_scopes(&self.curr_scope_chain),
+                                &item.name,
+                                &item.namespace,
+                            ) {
+                                Ok(res) => res,
+                                Err(err) => {
+                                    self.errors.push(err);
+                                    continue;
+                                }
+                            }
+                        }
+                    };
+
+                    let scope = self.current_scope_mut();
+
+                    let resolved_item_id = match resolved_item {
+                        Res::Item(
+                            ItemId {
+                                package: Some(_), ..
+                            },
+                            _,
+                        ) => {
+                            self.errors.push(Error::ExportedExternalItem(item.span));
+                            continue;
+                        }
+                        Res::Item(id, _) => id,
```

Comment:
Feels like you should use the `ItemStatus` in some way to determine whether the item is available in the current configuration.

e.g. the error message in the below export could be a little more helpful (just like it is in the function call)

```qsharp
namespace Foo {
    export { BaseOp }; // "this is not a callable or type", no help
             ^^^^^^

    @EntryPoint()
    operation Main() : Unit {
        BaseOp(); // help: found a matching item `Foo.BaseOp` that is not available for the current compilation configuration
        ^^^^^^
    }

    @Config(Base)
    operation BaseOp() : Unit {

    }
}
```